packages/amplify_auth_cognito/example/lib/Widgets/SignInWidget.dart

```diff
@@ -167,21 +166,24 @@ class _SignInWidgetState extends State<SignInWidget> {
                         height: 2,
                         color: Colors.deepPurpleAccent,
                       ),
-                      onChanged: (AuthProvider newValue) {
+                      onChanged: (AuthProvider? newValue) {
                         setState(() {
-                          provider = newValue;
+                          provider = newValue!;
```

Comment:
Agreed.  `newValue` cannot actually be null.  I think this case is a bit weird because as you can see we only provided non null AuthProviders to the items field of the overarching `Container` class here: 
```

items: <AuthProvider>[
                        AuthProvider.google,
                        AuthProvider.facebook,
                        AuthProvider.amazon
```

So the `AuthProvider? newValue` in `onChanged` can't actually be null, but we have to say it is nullable because the `Container` class defines its `onChanged` method with that nullable input type.

Reply:
It seems that if the `onChanged` method of the `DropdownButton` has a nullable value as its signature, then there might be a way to interact with the UI element which results in `newValue` being undefined (nothing selected somehow?) and therefore `null`.

I'm not really sure why this would be the case though or if this component is just being a little extra cautious about asking you to enforce null safety. If we're completely certain it's impossible for `newValue` to be passed a `null` value by the component then the casting is all good.

Reply:
Yeah I assumed it was being a little extra cautious but I'm not sure about that.  I think what you're suggesting is right and we might as well add a null check to be consistent and safe.